src/compiler/crystal/macros/methods.cr

```diff
@@ -211,11 +211,46 @@ module Crystal
         run_args << @last.to_macro_id
       end
 
-      success, result = @program.macro_run(filename, run_args)
-      if success
-        @last = MacroId.new(result)
+      result = @program.macro_run(filename, run_args)
+      if result.status.success?
+        @last = MacroId.new(result.stdout)
       else
-        node.raise "Error executing run: #{original_filename} #{run_args.map(&.inspect).join " "}\n\nGot:\n\n#{result}\n"
+        command = "#{original_filename} #{run_args.map(&.inspect).join " "}"
+
+        message = IO::Memory.new
+        message << "Error executing run (exit code: #{result.status.exit_code}): #{command}\n"
+
+        if result.stdout.empty? && result.stderr.empty?
+          message << "\nGot no output."
+        else
+          message << "\e[0m" if Colorize.enabled?
```

Comment:
for `"abc".colorize.reset`, should I save the current formatting, reset, put `"abc"`, then restore the formatting? Or should I go simple for this PR, and just reset and put the text?